README.md

```diff
@@ -72,19 +72,39 @@ See [action.yml](action.yml) for the full documentation for this action's inputs
 ## Credentials
 
 We recommend following [Amazon IAM best practices](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html) for the AWS credentials used in GitHub Actions workflows, including:
-* Do not store credentials in your repository's code.  You may use [GitHub Actions secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets) to store credentials and redact credentials from GitHub Actions workflow logs.
-* [Create an individual IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#create-iam-users) with an access key for use in GitHub Actions workflows, preferably one per repository. Do not use the AWS account root user access key.
+* Do not store credentials in your repository's code.
 * [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege) to the credentials used in GitHub Actions workflows.  Grant only the permissions required to perform the actions in your GitHub Actions workflows.
-* [Rotate the credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#rotate-credentials) used in GitHub Actions workflows regularly.
 * [Monitor the activity](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#keep-a-log) of the credentials used in GitHub Actions workflows.
 
 ## Assuming a Role
-If you would like to use the static credentials you provide to this action to assume a role, you can do so by specifying the role ARN in `role-to-assume`.
-The role credentials will then be configured in the Actions environment instead of the static credentials you have provided.
-The default session duration is 6 hours, but if you would like to adjust this you can pass a duration to `role-duration-seconds`.
+We recommend using GitHub's OIDC provider to get short-lived credentials needed for your actions. 
+Specifying `role-to-assume` without providing an `aws-access-key-id` or a `web-identity-token-file` will signal to the action that you wish to use the OIDC provider.
+The default session duration is 1 hour when using the OIDC provider to directly assume an IAM Role.
```

Comment:
I may be misunderstanding this documentation, but it doesn't look like this logic was implemented in the PR and the default is still using `MAX_ACTION_RUNTIME` regardless of method. I think 3600 seconds is the default on new roles and that would mean we'd need to set `role-duration-seconds` on every use of the plugin or change the maximum session duration on our roles. I like the 1 hour default if using OIDC.

I'm really excited for this to land and it's been working great in testing.

Reply:
This was a really good catch! Thank you! I'm fixing it now. It'll default to 3600 for GH OIDC provider Roles and remain 6hrs (for backwards compatibility reasons) for the rest.

Reply:
That sounds perfect, thanks!

Reply:
Fixed, and I updated the test to validate it.